libcodechecker/libhandlers/analyze.py

```diff
@@ -289,11 +302,18 @@ def main(args):
         with open(args.skipfile, 'r') as skipfile:
             metadata['skip_data'] = [l.strip() for l in skipfile.readlines()]
 
+    # Update metadata dictionary with old values.
+    metadata_file = os.path.join(args.output_path, 'metadata.json')
```

Comment:
There is no "incompatibility", the `store` command works without `metadata.json` if we define `works` as `puts the bugs found in the folder into the database`.

The pre-`5.8` `plist` command, as *update* mode was not supported, badly worked. It was expected from the user to always store a directory of plists into an empty run.

Metadata JSON is needed to connect `plist` files to source files and source build actions, so that the current storage system (on `master`) knows when to eliminate bugs linked to a `sourcefile, buildaction) pair. With the onset of #709, #724 and the possible upcoming schema changes, the proper requirement on having a `metadata.json` will be revised.